plugin-server/src/worker/ingestion/app-metrics.ts

```diff
@@ -61,52 +62,43 @@ const safeJSONStringify = configure({
 })
 
 export class AppMetrics {
-    hub: Hub
+    kafkaProducer: KafkaProducerWrapper
     queuedData: Record<string, QueuedMetric>
 
     flushFrequencyMs: number
+    maxQueueSize: number
 
-    timer: NodeJS.Timeout | null
+    lastFlushTime: number
+    // For quick access to queueSize instead of using Object.keys(queuedData).length every time
+    queueSize: number
```

Comment:
nit: This adds an extra attribute that we have to keep track on with every method that manipulates the queue. Should be fine for now, but it could be bug-prone as it introduces a problem we didn't have: Forgetting to update the queueSize.

If this was Python I would suggest using a property that does the equivalent `Object.keys(queuedData).length` call, e.g.:

```python
@property
def queue_size(self):
    return len(self.queuedData)
```

But with js I'm not sure if there's anything better.

The issue grows only if we add new methods that manipulate the queue, so when we do that we can tackle this too.